src/azul/service/manifest_service.py

```diff
@@ -891,19 +902,23 @@ def write_to(self, output: IO[str]) -> Optional[str]:
         for hit in self._create_request().scan():
             doc = self._hit_to_doc(hit)
             assert isinstance(doc, dict)
+            file_ = one(doc['contents']['files'])
             # FIXME: The slice is a hotfix. Reconsider.
             #        https://github.com/DataBiosphere/azul/issues/2649
             for bundle in list(doc['bundles'])[0:100]:  # iterate over copy …
                 doc['bundles'] = [bundle]  # … to facilitate this in-place modification
                 row = {}
                 for doc_path, column_mapping in self.manifest_config.items():
-                    entities = self._get_entities(doc_path, doc)
+                    entities = [
+                        dict(e, file_url=self._azul_file_url(file_))
```

Comment:
It looks like `_azul_file_url` is being called multiple times with the same argument. The return value should be factored into a variable.